python/gps/algorithm/algorithm_badmm.py

```diff
@@ -34,19 +34,39 @@ def __init__(self, hyperparams):
             self._hyperparams['policy_opt'], self.dO, self.dU
         )
 
-    def iteration(self, sample_lists):
+    def iteration(self, sample_lists, itr=0):
         """
         Run iteration of BADMM-based guided policy search.
 
         Args:
             sample_lists: List of SampleList objects for each condition.
         """
+
+        if self._hyperparams['ioc']:
+            self.N = sum(len(self.sample_list[i]) for i in self.sample_list.keys())
+            self.num_samples = [len(self.sample_list[i]) for i in self.sample_list.keys()]
         for m in range(self.M):
             self.cur[m].sample_list = sample_lists[m]
 
-        self._set_interp_values()
-        self._update_dynamics()  # Update dynamics model using all sample.
-        self._update_step_size()  # KL Divergence step size.
+            if self._hyperparams['ioc']:
+                prev_samples = self.sample_list[m].get_samples()
+                prev_samples.extend(sample_lists[m].get_samples())
+                self.sample_list[m] = SampleList(prev_samples)
+                self.N += len(sample_lists[m])
+
+
```

Comment:
only one line between code chunks, not two. (not just here -- please look over other code too.)